maven.gradle

```diff
@@ -1,12 +1,19 @@
-import com.infinum.maven.SonatypeConfiguration
-
-SonatypeConfiguration.metaClass.constructor = { ->
-    def constructor = SonatypeConfiguration.class.getConstructor()
-    def instance = constructor.newInstance()
-    instance.load()
-    instance
+def properties = new Properties()
+def propsFile = file("$rootDir/publish.properties")
+if (propsFile.exists()) {
+    properties.load(propsFile.newInputStream())
 }
 
-ext {
-    sonatype = new SonatypeConfiguration()
-}
\ No newline at end of file
+ext.sonatype = [
+
+        url     : properties["sonatype.url"]
+                ?: "https://oss.sonatype.org/service/local/staging/deploy/maven2",
+
+        username: findProperty("sonatypeUsername")?.toString()
+                ?: properties["sonatype.user"]
+                ?: System.getenv("MAVEN_CENTRAL_USER"),
```

Comment:
1. use sonatypeUsername parameter from the command (if parameter is attached in the command) so you can deploy by doing: `./gradlew deploy -PsonatypeUsername={yourMavenUsername} -PsonatypePassword={yourMavenPassword}`.
2. use property defined in `publish.properties` file in the root directory. You can run just `./gradlew deploy` in case you have that file in the root of the project.
3. use global variable defined in the system. No need to create publish.properties file. You can run just `./gradlew deploy`.